process_email.py

```diff
@@ -622,35 +620,53 @@ def _handle_attachment(self, part, file_name, file_path):
                 del attach_meta_info['content']
                 curr_attach['should_ignore'] = True
 
-        part_payload = part.get_payload(decode=True)
-        try:
-            if not part_payload:
-                with open(file_path, 'wb'):
-                    self._base_connector.debug_print("Payload not found, hence adding empty file")
-            else:
+        if type(part.get_payload()) == list:
+            try:
+                part_payload = part.get_payload()[0]
+                self._debug_print("payload content type: {0}".format(type(part_payload)))
+                with open(file_path, 'wb') as f:
+                    f.write(part_payload.as_string().encode('utf-8'))
+            except IndexError as e:
+                error_message = self._base_connector._get_error_message_from_exception(e)
+                self._base_connector.debug_print("Error occurred while accessing attachment payload. Error Details: {}".format(error_message))
                 with open(file_path, 'wb') as f:
                     f.write(part_payload)
-        except IOError as e:
-            error_message = self._base_connector._get_error_message_from_exception(e)
+            except Exception as e:
+                error_message = self._base_connector._get_error_message_from_exception(e)
+                self._base_connector.debug_print("Error occurred while adding file to Vault. Error Details: {}".format(error_message))
+                return
+        else:
+            self._debug_print("this payload is a string")
+            part_payload = part.get_payload(decode=True)
+            self._debug_print("handle attach part payload 2 {0}".format(part_payload))
             try:
-                if "File name too long" in error_message:
-                    new_file_name = "ph_long_file_name_temp"
-                    file_path = "{}{}".format(file_path.rstrip(file_name.replace('<', '').replace('>', '').replace(' ', '')), new_file_name)
-                    self._base_connector.debug_print("Original filename: {}".format(file_name))
-                    self._base_connector.debug_print("Modified filename: {}".format(new_file_name))
-                    with open(file_path, 'wb') as uncompressed_file:
-                        uncompressed_file.write(part_payload)
+                if not part_payload:
+                    with open(file_path, 'wb'):
```

Comment:
We'd still have to call `.close()` so I think there is no reason not to use the context manager. I'd move the `if` inside of it though instead of the other way around

Reply:
Makes sense moving the if inside